src/tools/clippy/clippy_utils/src/ty/mod.rs

```diff
@@ -326,8 +328,14 @@ pub fn has_drop<'tcx>(cx: &LateContext<'tcx>, ty: Ty<'tcx>) -> bool {
 // Returns whether the type has #[must_use] attribute
 pub fn is_must_use_ty<'tcx>(cx: &LateContext<'tcx>, ty: Ty<'tcx>) -> bool {
     match ty.kind() {
-        ty::Adt(adt, _) => cx.tcx.has_attr(adt.did(), sym::must_use),
-        ty::Foreign(did) => cx.tcx.has_attr(*did, sym::must_use),
+        ty::Adt(adt, _) => find_attr!(
```

Comment:
All the `find_attr!`s with `cx.tcx.get_all_attrs` are a little ugly, is there a nicer way to do this?
If not we could consider making a new macro for that

Reply:
if only there were associated macros... at least this is easy to find now to fix later but I'm open for suggestions for better approaches.

Reply:
Could still get away with a `find_attr_ctx!(cx.tcx, adr.did(), ...)`, but not sure how big of an improvement that is.
This is also fine for now, definitely no need to block the PR on this